### ISSUE TITLE
Killing plugin - Failed to parse

### ISSUE THREAD

**Issue by waffi121** [Ride-The-Lightning/c-lightning-REST#101]
C-Lightning Version: v0.10.2
c-lighting-REST: v0.6.1
RTL:  v0.12.2

`2022-03-26T05:06:05.393Z UNUSUAL plugin-plugin.js: Killing plugin: Failed to parse JSON response 'SyntaxError: Unexpected token { in JSON at position 126?    at JSON.parse (<anonymous>)?    at Plugin._mainLoop (/home/btc/c-lightning-REST-0.6.1/node_modules/clightningjs/src/plugin.js:164:20)?    at Socket.<anonymous> (/home/btc/c-lightning-REST-0.6.1/node_modules/clightningjs/src/plugin.js:209:12)?    at Socket.emit (node:events:526:28)?    at emitReadable_ (node:internal/streams/readable:578:12)?    at processTicksAndRejections (node:internal/process/task_queues:82:21)?SyntaxError: Unexpected token { in JSON at position 126?    at JSON.parse (<anonymous>)?    at Plugin._mainLoop (/home/btc/c-lightning-REST-0.6.1/node_modules/clightningjs/src/plugin.js:164:20)?    at Socket.<anonymous> (/home/btc/c-lightning-REST-0.6.1/node_modules/clightningjs/src/plugin.js:209:12)?    at Socket.emit (node:events:526:28)?    at emitReadable_ (node:internal/streams/readable:578:12)?    at processTicksAndRejections (node:internal/process/task_queues:82:21)?SyntaxError: Unexpected token { in JSON at position 126?    at JSON.parse (<anonymous>)?    at Plugin._mainLoop (/home/btc/c-lightning-REST-0.6.1/node_modules/clightningjs/src/plugin.js:164:20)?    at Socket.<anonymous> (/home/btc/c-lightning-REST-0.6.1/node_modules/clightningjs/src/plugin.js:209:12)?    at Socket.emit (node:events:526:28)?    at emitReadable_ (node:internal/streams/readable:578:12)?    at processTicksAndRejections (node:internal/process/task_queues:82:21)?SyntaxError: Unexpected token { in JSON at position 126?    at JSON.parse (<anonymous>)?    at Plugin._mainLoop (/home/btc/c-lightning-REST-0.6.1/node_modules/clightningjs/src/plugin.js:164:20)?    at Socket.<anonymous> (/home/btc/c-lightning-REST-0.6.1/node_modules/clightningjs/src/plugin.js:209:12)?    at Socket.emit (node:events:526:28)?    at emitReadable_ (node:internal/streams/readable:578:12)?    at processTicksAndRejections (node:internal/process/task_queues:82:21)?{"jsonrpc":"2.0","method":"log","params":{"level":"warn","message":"Client disconnected, total clients: 0"}}'`

**Comment by saubyk** [comment]
Hi @waffi121 can you please provide more detail on the scenario where you getting this error?

**Comment by saubyk** [comment]
Hi @waffi121 can you please provide more detail on the scenario where you getting this error?

**Comment by waffi121** [comment]
unfortunately, I cannot reproduce it exactly.

Yesterday restarted everything, RTL never opened -> no errorr
Today RTL opened and again the same error. 

helps the log from rtl? 

`ERROR: GetInfo => Get Info Error: {"name":"RequestError","message":"Error: connect ECONNREFUSED 127.0.0.1:3001","cause":{"errno":-111,"code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":3001},"error":{"errno":-111,"code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":3001},"options":{"url":"https://localhost:3001/v1/getinfo","rejectUnauthorized":false,"json":true,"headers":{},"method":"GET","qs":{},"simple":true,"resolveWithFullResponse":false,"transform2xxOnly":false}}
`

**Comment by dzdidi** [comment]
Having the same issue, whic appears to be caused by plugin crash for the reason described here https://github.com/Ride-The-Lightning/c-lightning-REST/issues/89 
I am trying to reproduce it and will post updates here. 
Some additional info, there is also clboss plugin, so far my theory is that it is clboss which causes cl to reply with data which breaks clREST

**Comment by dzdidi** [comment]
here is the log input for message which caused crash:
```
{"jsonrpc":"2.0","method":"forward_event","params":{"forward_event":{"payment_hash":"5***************************************************************","in_channel":"************","out_channel":"************","in_msatoshi":********,"in_msat":"********msat","out_msatoshi":********,"out_msat":"********msat","fee":****,"fee_msat":"****msat","status":"failed","received_time":1650728930.700,"resolved_time":1650728937.504}}}??{"jsonrpc":"2.0","method":"forward_event","params":{"forward_event":{"payment_hash":"****************************************************************","in_channel":"************","out_channel":"************","in_msatoshi":********,"in_msat":"********msat","out_msatoshi":********,"out_msat":"********msat","fee":****,"fee_msat":"****msat","status":"failed","received_time":1650728931.439,"resolved_time":1650728937.507}}}
```
So clightning gives two messages in a single chunk  which makes cl-rest err on json parse. It seemed to be fixed in the dependency so we just need to update it https://github.com/lightningd/clightningjs/blob/master/src/plugin.js#L178

**Comment by BTCBellyButton** [comment]
Similar issue here.
Bitcoin:23.0
CLN version: 0.11.0.1
c-lighting-REST: v0.7.0

![Screenshot from 2022-04-27 19-44-08](https://user-images.githubusercontent.com/68573637/165654077-8f7ca05b-6048-463d-88b5-3f148b2d809a.png)

C-Lightning log:

> 2022-04-28T00:31:54.937Z UNUSUAL plugin-plugin.js: Killing plugin: Failed to parse JSON response 'SyntaxError: Unexpected token { in JSON at position 206?    at JSON.parse (<anonymous>)?    at Plugin._mainLoop (/home/go/c-lightning-REST/node_modules/clightningjs/src/plugin.js:164:20)?    at Socket.<anonymous> (/home/go/c-lightning-REST/node_modules/clightningjs/src/plugin.js:209:12)?    at Socket.emit (node:events:526:28)?    at emitReadable_ (node:internal/streams/readable:578:12)?    at processTicksAndRejections (node:internal/process/task_queues:82:21)?SyntaxError: Unexpected token { in JSON at position 203?    at JSON.parse (<anonymous>)?    at Plugin._mainLoop (/home/go/c-lightning-REST/node_modules/clightningjs/src/plugin.js:164:20)?    at Socket.<anonymous> (/home/go/c-lightning-REST/node_modules/clightningjs/src/plugin.js:209:12)?    at Socket.emit (node:events:526:28)?    at emitReadable_ (node:internal/streams/readable:578:12)?    at processTicksAndRejections (node:internal/process/task_queues:82:21)?{"jsonrpc":"2.0","method":"log","params":{"level":"warn","message":"Client disconnected, total clients: 0"}}'

**Comment by dzdidi** [comment]
@BTCBellyButton this should be resolved in 0.7.1 (not yet release, as far as I know)

**Comment by saubyk** [comment]
merged to release branch
